encoding/thrift/inbound.go

```diff
@@ -47,41 +47,12 @@ type thriftOnewayHandler struct {
 }
 
 func (t thriftUnaryHandler) Handle(ctx context.Context, treq *transport.Request, rw transport.ResponseWriter) error {
-	if err := errors.ExpectEncodings(treq, Encoding); err != nil {
-		return err
-	}
-
-	ctx, call := encodingapi.NewInboundCall(ctx)
-	if err := call.ReadFromRequest(treq); err != nil {
-		return err
-	}
-
-	buf := bufferpool.Get()
-	defer bufferpool.Put(buf)
-	if _, err := buf.ReadFrom(treq.Body); err != nil {
-		return err
-	}
-
-	// We disable enveloping if either the client or the transport requires it.
-	proto := t.Protocol
-	if !t.Enveloping {
-		proto = disableEnvelopingProtocol{
-			Protocol: proto,
-			Type:     wire.Call, // we only decode requests
-		}
-	}
-
-	envelope, err := proto.DecodeEnveloped(bytes.NewReader(buf.Bytes()))
+	ctx, call, reqValue, responder, err := handleRequest(ctx, treq, wire.Call, t.Protocol, t.Enveloping)
```

Comment:
optional: this could be reduced to three arguments if you made handleRequest a method

Reply:
Would be nice, but sharing this method between thriftUnaryHandler and thriftOnewayHandler, doesn’t have a common receiver type.